### ISSUE TITLE
Can't start server: backend error TypeError: Cannot read property 'toLowerCase' of undefined

### ISSUE THREAD

**Issue by aneilbaboo** [sysgears/apollo-universal-starter-kit#532]
I'm trying to `yarn watch` with the following settings, and receiving various errors and warnings:

npm --version: 5.5.1
yarn --version: 1.3.2
node --version: v8.1.4
commit: 0e8676f0d3ba6aae311e24419daffddf9e859a54

```shell
➜ yarn  watch
yarn run v1.3.2
$ spin watch
webpack-for-webDll info Generating vendor_web DLL bundle with modules:
["antd","apollo-cache-inmemory","apollo-client","apollo-fetch","apollo-fetch-upload","apollo-link","apollo-link-batch-http","apollo-link-ws","apollo-logger","bootstrap","error-stack-parser","graphql","graphql-auth","graphql-iso-date","graphql-tag","history","immutability-helper","jwt-decode","lodash","minilog","persistgraphql","prop-types","react","react-apollo","react-cookie","react-debounce-input","react-dom","react-dropzone","react-ga","react-helmet","react-hot-loader","react-native-web","react-redux","react-router","react-router-dom","react-router-redux","react-transition-group","reactstrap","redux","redux-devtools-extension","redux-form","styled-components","subscriptions-transport-ws"]
webpack-for-webDll Time: 12947ms
[HPM] Proxy created: !/*.hot-update.{json,js}  ->  http://localhost:8080
webpack-for-web Webpack web dev server listening on http://localhost:3000
webpack-for-server Starting backend
webpack-for-server Time: 19131ms
webpack-for-server Spawning node --inspect /Users/aneil/code/precisely/apollo-universal-starter-kit/build/server/index.js
Debugger listening on ws://127.0.0.1:9229/7923f59a-00e4-46c2-9c06-a15de3f28838
For help see https://nodejs.org/en/docs/inspector
webpack-for-web debug Webpack dev server is waiting for backend at localhost:8080
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
backend info API is now running on port 8080
webpack-for-web debug Backend has been started, resuming webpack dev server...
cat: /proc/self/cgroup: No such file or directory
webpack-for-web bundle is now VALID.
webpack-for-web Time: 6529ms
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
    in CounterView
    in Counter
    in Apollo(Counter)
    in Apollo(Apollo(Counter))
    in Connect(Apollo(Apollo(Counter)))
    in Route
    in Switch
    in Router
    in StaticRouter
    in QueryRecyclerProvider
    in ApolloProvider
    in Provider
    in CookiesProvider
    in App (at renderApplication.js:36)
    in div
    in View (at AppContainer.js:52)
    in div
    in View (at AppContainer.js:51)
    in AppContainer (at renderApplication.js:35)
    in StyleSheetManager
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
    in CounterView
    in Counter
    in Apollo(Counter)
    in Apollo(Apollo(Counter))
    in Connect(Apollo(Apollo(Counter)))
    in Route
    in Switch
    in Router
    in StaticRouter
    in QueryRecyclerProvider
    in ApolloProvider
    in Provider
    in CookiesProvider
    in App (at renderApplication.js:36)
    in div
    in View (at AppContainer.js:52)
    in div
    in View (at AppContainer.js:51)
    in AppContainer (at renderApplication.js:35)
    in StyleSheetManager
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
    in CounterView
    in Counter
    in Apollo(Counter)
    in Apollo(Apollo(Counter))
    in Connect(Apollo(Apollo(Counter)))
    in Route
    in Switch
    in Router
    in StaticRouter
    in QueryRecyclerProvider
    in ApolloProvider
    in Provider
    in CookiesProvider
    in App (at renderApplication.js:36)
    in div
    in View (at AppContainer.js:52)
    in div
    in View (at AppContainer.js:51)
    in AppContainer (at renderApplication.js:35)
    in StyleSheetManager
backend error TypeError: Cannot read property 'toLowerCase' of undefined
    at ReactDOMServerRenderer.renderDOM (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2287:27)
    at ReactDOMServerRenderer.render (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2281:21)
    at ReactDOMServerRenderer.read (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2217:19)
    at Object.renderToString (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2484:25)
    at _callee$ (/Users/aneil/code/precisely/apollo-universal-starter-kit/src/server/middleware/website.jsx:87:1)
    at tryCatch (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:299:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:117:21)
    at step (/Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/aneil/code/precisely/apollo-universal-starter-kit/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)

```

**Comment by aneilbaboo** [comment]
The error happens in ReactDOMServerRenderer#renderDOM. It's expecting element.type to be defined. I added some logging:

```javascript
ReactDOMServerRenderer.prototype.renderDOM = function renderDOM(element, context, parentNamespace) {
    console.log("ELEMENT=", element);  // I added this
    var tag = element.type.toLowerCase();  // this line throws the error
    ...
```

The console statement produces this:
```shell
ELEMENT= { '$$typeof': Symbol(react.element),
  type: undefined,
  key: null,
  ref: null,
  props: { children: [ [Object], [Object], [Object] ] },
  _owner: null,
  _store: {} }
backend error TypeError: Cannot read property 'toLowerCase' of undefined
    at ReactDOMServerRenderer.renderDOM (/Users/aneil/code/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2289:28)
    at ReactDOMServerRenderer.render (/Users/aneil/code/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2281:21)
    at ReactDOMServerRenderer.read (/Users/aneil/code/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2217:19)
    at Object.renderToString (/Users/aneil/code/apollo-universal-starter-kit/node_modules/react-dom/cjs/react-dom-server.node.development.js:2486:25)
    at _callee$ (/Users/aneil/code/apollo-universal-starter-kit/src/server/middleware/website.jsx:87:1)
    at tryCatch (/Users/aneil/code/apollo-universal-starter-kit/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/aneil/code/apollo-universal-starter-kit/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:299:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/aneil/code/apollo-universal-starter-kit/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:117:21)
    at step (/Users/aneil/code/apollo-universal-starter-kit/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/aneil/code/apollo-universal-starter-kit/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by vlasenko** [comment]
@aneilbaboo I cannot reproduce any of your issues
```
bash-3.2$ node --version
v8.1.4
bash-3.2$ yarn --version
1.3.2
bash-3.2$ npm --version
5.5.1
```

**Comment by mitjade** [comment]
@vlasenko `toLowerCase` is called inside `tools/cli/module.js`. Should't this be excluded in this case?

**Comment by vlasenko** [comment]
@mitjade Here it is called inside `react-dom-server.node.development.js`

**Comment by mitjade** [comment]
@vlasenko Sorry, just quickly glanced at this and remembered I use this somewhere.

**Comment by aneilbaboo** [comment]
Hmmm. Thank you for looking into this, @vlasenko. I just tried on a different machine (also macOS), and it worked. Looks like a problem on my machine. Sorry to waste your time. Closing this now.

**Comment by aneilbaboo** [comment]
Just in case anyone else runs into this - I finally tracked down this nasty problem.  It was due to `.babelrc` in a parent folder. (It seems related to  https://github.com/babel/ember-cli-babel/issues/92)

Specifically, I had the following .babelrc in my `~/code` folder:
```
{
  "presets": [
    "es2015",
    "stage-0",
    "react"
  ]
}
```

Attempting to `yarn watch` when the kit is in any (even deeply nested) subdirectory, will result in an error due to jsx files not being properly processed. For example, attempting to require any of the jsx files in `src/client/modules/common/components/web/ui-bootstrap/components` gives an undefined value.
